code/curate_filename.py

```diff
@@ -0,0 +1,51 @@
+import os
+import argparse
+import pandas as pd
+import re
+
+def process_filename(original_path):
+    """Extracts and modifies the filename path based on given rules."""
```

Comment:
Maybe add an example?

```suggestion
    """
    Extracts and modifies the filename path based on given rules.
    Specifically, from the absolute input path, keep only `sub-XXX/anat/sub-XXX_T2w_label-SC_seg.nii.gz`.
    """
```